src/main/java/winstone/AbstractSecuredConnectorFactory.java

```
@@ -122,44 +117,30 @@ protected void configureSsl( Map args, Server server ) throws IOException
 
-            BigInteger mod, privExpo;
-            try {
-                Class<?> disC = Class.forName( "sun.security.util.DerInputStream" );
-                Object dis = disC.getConstructor( byte[].class ).newInstance( (Object) baos.toByteArray() );
-                Object[] seq = (Object[]) disC.getMethod( "getSequence", int.class ).invoke( dis, 0 );
-                Method getBigInteger = seq[0].getClass().getMethod( "getBigInteger" );
-                // int v = seq[0].getInteger();
-                mod = (BigInteger) getBigInteger.invoke( seq[1] );
-                // pubExpo
-                // p1, p2, exp1, exp2, crtCoef
-                privExpo = (BigInteger) getBigInteger.invoke( seq[3] );
-            } catch ( Exception x ) {
-                throw new WinstoneException( SSL_RESOURCES.getString( "HttpsConnectorFactory.LoadPrivateKeyError" ), x );
-            }
-            Logger.log( Level.WARNING, SSL_RESOURCES, "HttpsConnectorFactory.LoadPrivateKey" );
+        PEMParser pemParser = new PEMParser(reader);
+        PKCS8EncodedKeySpec privateKey = null;
+
+        try {
+            PEMKeyPair pemKeyPair = (PEMKeyPair) pemParser.readObject();
+            privateKey = new PKCS8EncodedKeySpec(pemKeyPair.getPrivateKeyInfo().getEncoded());
+        }
+        catch (IOException ioex) {
+            Logger.log(Level.WARNING, SSL_RESOURCES, ioex.getMessage());
+        }
 
-            KeyFactory kf = KeyFactory.getInstance( "RSA" );
-            return kf.generatePrivate( new RSAPrivateKeySpec( mod, privExpo ) );
+        PrivateKey privateKeyOutput = null;
+        try {
+            privateKeyOutput = KeyFactory.getInstance("RSA").generatePrivate(privateKey);
+        } catch (InvalidKeySpecException e) {
+            Logger.log(Level.WARNING, SSL_RESOURCES, e.getMessage());
+        } catch (NoSuchAlgorithmException e) {
+            Logger.log(Level.WARNING, SSL_RESOURCES, e.getMessage());
         }
+
+        return  privateKeyOutput;
```

as we ignore errors. we will return null?